### PR TITLE
Add `Char#printable?`

### PR DESCRIPTION
**Pull request by straight-shoota**
This method defines whether a character is printable.

Preparation for #11406

### CODE REVIEW

File: src/char.cr [hunk 1]
```diff
@@ -489,11 +489,23 @@ struct Char
...
 
-  # Returns `true` if this is char is a mark character according to unicode.
+  # Returns `true` if this char is a mark character according to unicode.
   def mark? : Bool
     Unicode.mark?(self)
   end
 
+  # Returns `true` if this char is a printable character.
+  #
+  # There is no universal definition of printable charactesr in Unicode.
```
Comment (Fryguy) [inline src/char.cr:499]:
```suggestion
  # There is no universal definition of printable characters in Unicode.
```

File: src/char.cr [hunk 2]
```diff
@@ -489,11 +489,23 @@ struct Char
...
+  # For the purpose of this method, all characters with a visible glyph and the
+  # ASCII whitesapce (` `) are considered printable.
```
Comment (Fryguy) [inline src/char.cr:501]:
```suggestion
  # ASCII whitespace (` `) are considered printable.
```

File: spec/std/char_spec.cr [hunk 3]
```diff
@@ -390,6 +390,15 @@ describe "Char" do
     '酒'.ascii?.should be_false
   end
 
+  it "#printable?" do
+    ' '.printable?.should be_true
+    'a'.printable?.should be_true
+    '酒'.printable?.should be_true
+    '\n'.printable?.should be_false
```
Comment (Sija) [inline spec/std/char_spec.cr:397]: I'd think that the newline character is included. It's certainly _visible_, and _printable_ - same as ASCII whitespace.

Comment (Fryguy) [reply spec/std/char_spec.cr:397]: I was thinking similar for tab.  I guess it depends what the purpose of this method is apart from what's needed in #11406

File: src/char.cr [hunk 4]
```diff
@@ -489,11 +489,23 @@ struct Char
...
+  # Returns `true` if this char is a printable character.
+  #
+  # There is no universal definition of printable characters in Unicode.
+  # For the purpose of this method, all characters with a visible glyph and the
+  # ASCII whitespace (` `) are considered printable.
+  #
+  # This means characters which are `control?` or `whitespace?` (except for ` `)
+  # are non-printable.
+  def printable?
+    !control? && (!whitespace? || self == ' ')
```
[the remark this answers is not in the document]
Comment (straight-shoota) [reply src/char.cr:506]: I don't follow. Unicode characters in any other category than control or whitespace are surely printable.

**Comment by HertzDevil** [comment]
This definition coincides with Go's (own) definiton, by the way: https://pkg.go.dev/unicode#IsPrint

**Comment by straight-shoota** [comment]
@HertzDevil Yes, that and Python's [`str.isprintable`](https://docs.python.org/3/library/stdtypes.html#str.isprintable) have served as inspiration.

File: src/char.cr (hunk 4 above)
[the remark this answers is not in the document]
Comment (j8r) [reply src/char.cr:506]: Yes my bad, what we want is unicode printable, not ascii printable.
Usually printable is more related to ASCII, I misread the intentions.

**Review by jhass — APPROVED** [review]

**Review by oprypin — APPROVED** [review]

**Pull request merged**